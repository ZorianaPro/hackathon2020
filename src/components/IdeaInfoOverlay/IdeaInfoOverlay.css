.IdeaInfoOverlay .IdeaInfoOverlay-Title {
    text-align: center;
    font-size: 2.2em;
    padding-top: 1em;
}
.IdeaInfoOverlay .IdeaInfoOverlay-Team .IdeaInfoOverlay-Team-Title {
    text-align: center;
    font-size: 1.5em;
    padding-top: 1em;
}

.IdeaInfoOverlay .IdeaInfoOverlay-Description {
    margin: 2em 0;
}
