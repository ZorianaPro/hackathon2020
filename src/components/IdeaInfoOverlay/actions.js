export const actions = {
	open: 'OPEN:OVERLAY',
	close: 'CLOSE:OVERLAY'
};

export const open = (ideaInfo) => ({
	type: actions.open,
	value: ideaInfo
});

export const close = () => ({
	type: actions.close,
	value: false
});

export default {
	open,
	close
};
