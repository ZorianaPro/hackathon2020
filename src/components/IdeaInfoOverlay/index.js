import React, { useState, useEffect, useCallback } from 'react';
import { useSelector, useDispatch } from "react-redux";
import Team from '../Team';
import './IdeaInfoOverlay.css';
import '../Overlay/Overlay.css';
import { close } from './actions';

const IdeaInfoOverlay = ({
	team,
  dispatch,
	isOpen,
	content
}) => {

	dispatch = useDispatch();

	[isOpen, content] = useSelector(state => [
		state.ideaInfoOverlayReducer.open,
		state.ideaInfoOverlayReducer.content
	]);

	const closeOverlay = () => {
		dispatch(close())
	};

	return (
		<div className="IdeaInfoOverlay">
			{ isOpen
				&& <div className="Overlay" key={`idea-${content.id}-overlay`}>
					<div className="Overlay-Container">
						<div className="Overlay-Box">
							<div className="Overlay-Close" onClick={() => closeOverlay()}>
								close
							</div>
							<div className="IdeaInfoOverlay-Title">
								{content.name}
							</div>
							<div className="IdeaInfoOverlay-Description">
								{content.description}
							</div>
							<div className="IdeaInfoOverlay-Team">
								<div className="IdeaInfoOverlay-Team-Title">
									Team
								</div>
							<Team full={ true }
							      team={ content.team }/>
							</div>
						</div>
					</div>
				</div>
			}
		</div>
	);
};

export default IdeaInfoOverlay;