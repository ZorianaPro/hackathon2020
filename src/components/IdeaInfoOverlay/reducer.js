import { actions } from "./actions";

const initialState = {
	open: false,
	ideaContent: {
		id: '',
		name: '',
		description: '',
		team: {}
	}
};

export default (state = initialState, action) => {
	switch (action.type) {
		case actions.open:
			return {
				open: true,
				ideaContent: action.value
			};
		case actions.close:
			return {
				open: false,
				ideaContent: initialState.ideaContent
			};
		default:
			return state;
	}
};
