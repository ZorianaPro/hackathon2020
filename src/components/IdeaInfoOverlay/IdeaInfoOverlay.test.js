import React from 'react';
import { shallow, mount } from 'enzyme';
import IdeaInfoOverlay from './';
import configureStore from "redux-mock-store";
import { Provider } from 'react-redux';

describe('IdeaInfoOverlay', () => {
	const initialState = {
		ideaInfoOverlayReducer: {
			open: true,
			content: {
				id: '',
				name: '',
				description: '',
				team: {}
			}
		}
	};
	const mockStore = configureStore();
	let store;

	beforeEach(() => {
		store = mockStore(initialState);
	});

	it('renders without crashing', () => {
		expect(shallow.bind(shallow, <Provider store={store}><IdeaInfoOverlay/></Provider>)).not.toThrow();
	});

	it('renders as expected', () => {
		expect(mount(<Provider store={store}> <IdeaInfoOverlay /></Provider>)).toMatchSnapshot();
	});

	describe('click close button', () => {
		let overlay;

		beforeEach(() => {
			store = mockStore(initialState);
			overlay = mount(<Provider store={store}> <IdeaInfoOverlay /></Provider>);
			overlay.find('.Overlay-Close').simulate('click');
		});
		it('closes overlay', () => {
			// expect(overlay.exists('.IdeaInfoOverlay .Overlay')).toBe(false);
		});
	});
});
