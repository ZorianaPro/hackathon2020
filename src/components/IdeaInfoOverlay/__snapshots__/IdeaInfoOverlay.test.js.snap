// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaInfoOverlay renders as expected 1`] = `
<IdeaInfoOverlay
  content={
    Object {
      "description": "",
      "id": "",
      "name": "",
      "team": Object {},
    }
  }
>
  <Overlay
    onClose={[Function]}
  >
    <div
      className="Overlay Overlay--isOpening"
    >
      <div
        className="Overlay-Content"
        style={
          Object {
            "animationDuration": "200ms",
          }
        }
      >
        <div
          className="Overlay-Mask"
          onClick={[Function]}
        />
        <div
          className="Overlay-Box"
        >
          <div
            className="Overlay-Close"
            onClick={[Function]}
          >
            close
          </div>
          <div
            className="IdeaInfoOverlay"
          >
            <p
              className="IdeaInfoOverlay-Title"
            />
            <p
              className="IdeaInfoOverlay-Description"
            />
            <div
              className="IdeaInfoOverlay-Team"
            >
              <div
                className="IdeaInfoOverlay-Team-Title"
              >
                Team
              </div>
              <Team
                full={true}
                team={Object {}}
              >
                <div
                  className="Team Team--full"
                >
                  <div
                    className="Team-Members"
                  >
                    <div
                      className="Team--empty"
                    >
                      No members yet
                    </div>
                  </div>
                </div>
              </Team>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Overlay>
</IdeaInfoOverlay>
`;
